pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go

```diff
@@ -825,7 +825,7 @@ func (sip *streamIngestionProcessor) flush() (*jobspb.ResolvedSpans, error) {
 			sip.metrics.CommitLatency.RecordValue(timeutil.Since(minBatchMVCCTimestamp.GoTime()).Nanoseconds())
 			sip.metrics.Flushes.Inc(1)
 			sip.metrics.IngestedBytes.Inc(int64(totalSize))
-			sip.metrics.SSTBytes.Inc(sip.batcher.GetSummary().SSTDataSize)
+			sip.metrics.SSTBytes.Inc(sip.batcher.GetBatchSummary().SSTDataSize)
 			sip.metrics.IngestedEvents.Inc(int64(len(sip.curKVBatch)))
```

Comment:
in a loop where we flush the range batcher but not the vanilla batcher, we'll double count the regular batcher by calling this anonymous function twice. I also don't even understand why all this incrementing logic even needs to be in an anonymous function. If you agree, could you simplify this logic, where we only increment the regular batcher metrics when we flush the regular batcher, and the same for the range batcher?

Reply:
I'm not sure I follow, when `flush` gets called we check which of the point key or range key batchers are empty and flush the non-empty ones. If there are only rangekeys we expect the size metrics to only contain the rangekey sizes. So things like `sip.curKVBatch` would be empty since we have seen no point keys. Similarly for if we only have point keys and no rangekeys we'd see SSTDataSize, logicalByteSize etc. only account for the point key. I've moved the logic into a helper method though. There was a bug where we were resetting the batcher before reading its summary, so sst bytes would always be incremented by 0.

Reply:
aha! thanks for the explanation. I didn't realize that `sip.curKV.size() == 0 iff batcher.GetBatchSummary().SSTDataSize "==" 0`. In general, it seems too much of the internals of the batcher is exposed here, so thank you for putting this stuff in a helper!

Reply:
thanks for pointing it out, helped me notice a bug!